src/utils/word.ts

```diff
@@ -1,74 +1,110 @@
-        });
-    };
+  return (word: string) => {
+    return Object.entries(indexToLetters).every(([i, letters]) => {
+      return letters.every((l) => word.includes(l) && word[i] !== l);
+    });
+  };
 };
 
 const includesAt = (indexToLetter: ToIncludeAt): Validator => {
-    return (word) => {
-        return Object.entries(indexToLetter).every(([i, l]) => word[i] === l);
-    };
+  return (word) => {
+    return Object.entries(indexToLetter).every(([i, l]) => word[i] === l);
+  };
 };
 
 export const getNextGuess = (
-    toExclude: ToExclude,
-    toIncludeNotAt: ToIncludeNotAt,
-    toIncludeAt: ToIncludeAt,
-    index: number,
-) => {    
-    const conditions: Validator[] = [
-        ...(index < 2 ? [hasAllUniqueLetters, isNotPlural] : []),
-        excludes(toExclude),
-        includesNotAt(toIncludeNotAt),
-        includesAt(toIncludeAt),
-    ];
-
-    const { word } = wordList
-        .reduce((max: { word: string, val: number } | null, word) => {
-            const shouldInclude = conditions.every(c => c(word));
-            if (!shouldInclude) return max;
-            const val = getWordValue(word);
-            return max?.val >= val ? max : { word, val };
-        }, null);
-    
-    return word;
+  toExclude: ToExclude,
+  toIncludeNotAt: ToIncludeNotAt,
+  toIncludeAt: ToIncludeAt,
+  index: number
+) => {
+  const conditions: Validator[] = [
+    ...(index < 2 ? [hasAllUniqueLetters, isNotPlural] : []),
+    excludes(toExclude),
+    includesNotAt(toIncludeNotAt),
+    includesAt(toIncludeAt),
+  ];
+
+  const { word } = wordList.reduce((max: { word: string; val: number } | null, word) => {
+    const shouldInclude = conditions.every((c) => c(word));
+    if (!shouldInclude) return max;
+    const val = getWordValue(word);
+    return max?.val >= val ? max : { word, val };
+  }, null);
+
+  return word;
+};
+
+/**
+ * Given a filename with a newline separated list of words, calculate each word's 'score' and return the resulting
+ * Object with each word as a key and its score as the value.
+ * @param wordListFilename string name of the file that contains words we want to score.
+ * @param outFilename string name of the file where the output should be saved
+ * @param wordScore (x: string) => number the function to use to score each word
+ * @returns Object<string, number> that maps a word to its score
+ */
+export const saveWordScores = (
+  wordListFilename: string = "sgb-words.txt",
+  outFilename: string = "sgb-word-scores.json",
+  wordScore: (word: string) => number = getWordValue
+): { [word: string]: number } => {
+  // if caller only provided a filename without a path (i.e. output.json) then
+  // default to looking up the input file in the 'data' directory of this project
+  if (!wordListFilename.includes("/")) {
+    wordListFilename = path.resolve(__dirname, `../../data/${wordListFilename}`);
+  }
+  const words = fs.readFileSync(wordListFilename).toString().split("\n");
```

Comment:
My buddy had trouble running this script on his Windows machine, until we realized that the newline character is slightly different on Windows vs. unix (see: https://stackoverflow.com/questions/1761051/difference-between-n-and-r/1761086). You can pass a regex that handles both though:

```suggestion
  const words = fs.readFileSync(wordListFilename).toString().split(/\r?\n|\r/);
```